src/test/java/com/microsoft/graph/authentication/TokenCredentialAuthProviderTest.java

```diff
@@ -0,0 +1,55 @@
+package com.microsoft.graph.authentication;
+
+import com.azure.core.credential.TokenCredential;
+import com.microsoft.graph.exceptions.AuthenticationException;
+import com.microsoft.graph.httpcore.IHttpRequest;
+import com.microsoft.graph.mocks.MockIHttpRequest;
+import com.microsoft.graph.mocks.MockTokenCredential;
+import com.microsoft.graph.options.HeaderOption;
+import okhttp3.Request;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+public class TokenCredentialAuthProviderTest {
+
+    public final String testToken = "CredentialTestToken";
+
+    @Test
+    public void TokenCredentialAuthProviderTestICoreAuthentication() throws AuthenticationException {
+
+        //Arrange
+        Request request = new Request.Builder().url("https://localhost").build();
```

Comment:
these tests are wrong, we should not be adding the bearer token on hostname other than graph.microsoft.com (or national deployments)

Reply:
fixed